lib/compat/wordpress-6.5/class-wp-navigation-block-renderer.php

```diff
@@ -197,8 +197,16 @@ private static function get_inner_blocks_from_navigation_post( $attributes ) {
 			// it encounters whitespace. This code strips it.
 			$compacted_blocks = gutenberg_block_core_navigation_filter_out_empty_blocks( $parsed_blocks );
 
+			// Run Block Hooks algorithm to inject hooked blocks.
+			$markup = gutenberg_hook_first_last_children_into_core_navigation( $compacted_blocks, $navigation_post->ID );
+			$blocks = parse_blocks( $markup );
```

Comment:
This is a sort of a public API thing ... not _really_ but now the markup in a post render hook of this block has the hooked blocks in it. Is this always going to be the case? Do all the other blocks behave the same or will it be gone when this block will be updated to work in a "normal" way?

Reply:
> This is a sort of a public API thing ... not really but now the markup in a post render hook of this block has the hooked blocks in it.

This is the desired outcome though -- for all practical purposes, hooked blocks are supposed to appear just like regular blocks to pretty much any filters/functions/… that come after.

> Is this always going to be the case? Do all the other blocks behave the same or will it be gone when this block will be updated to work in a "normal" way?

Any hooked blocks -- irrespective of their anchor block -- behave that way. I'm afraid I don't understand what you mean by 'will it be gone when this block will be updated to work in a "normal" way?' -- could you clarify? 😊

Reply:
It's fine if all hooked blocks behave like this, I was unsure of how it all works.